test/matrix_generator.cc

```diff
@@ -369,38 +375,29 @@ void generate_svd(
     }
 
     // random U, m-by-min_mn
```

Comment:
Yes, using `parallel` with `task` is a more flexible solution than collapsing 2 `for` loops.